frontend/src/pages/Home.jsx

```diff
@@ -14,7 +14,7 @@ function Home() {
   const navigate = useNavigate();
 
   const getToilets = useCallback(async () => {
-    if (token === undefined || token === null) {
+    if (token === 'undefined' || token === null) {
```

Comment:
Can we test whether `undefined` needs to be wrapped as a string as `undefined` is a JS data type so unless we're setting it to the string 'undefined' we shouldn't need to wrap it